examples/insurance/src/main/ml-modules/root/custom-modules/custom/enrichAdvantage/main.sjs

```diff
@@ -0,0 +1,114 @@
+  let context = content.context;
+   
+  //let's set our output format, so we know what we're exporting
+  let outputFormat = options.outputFormat ? options.outputFormat.toLowerCase() : datahub.flow.consts.DEFAULT_FORMAT;
+   
+  //here we check to make sure we're not trying to push out a binary or text document, just xml or json
+  if (outputFormat !== datahub.flow.consts.JSON && outputFormat !== datahub.flow.consts.XML) {
+    datahub.debug.log({
+      message: 'The output format of type ' + outputFormat + ' is invalid. Valid options are ' + datahub.flow.consts.XML + ' or ' + datahub.flow.consts.JSON + '.',
+      type: 'error'
+    });
+    throw Error('The output format of type ' + outputFormat + ' is invalid. Valid options are ' + datahub.flow.consts.XML + ' or ' + datahub.flow.consts.JSON + '.');
+  }
+   
+  /*
+  This scaffolding assumes we obtained the document from the database. If you are inserting information, you will
+  have to map data from the content.value appropriately and create an instance (object), headers (object), and triples
+  (array) instead of using the flowUtils functions to grab them from a document that was pulled from MarkLogic.
+  Also you do not have to check if the document exists as in the code below.
+   
+  Example code for using data that was sent to MarkLogic server for the document
+  let instance = content.value;
+  let triples = [];
+  let headers = {};
+   */
+   
+  //Here we check to make sure it's still there before operating on it
+  if (!fn.docAvailable(id)) {
+    datahub.debug.log({message: 'The document with the uri: ' + id + ' could not be found.', type: 'error'});
+    throw Error('The document with the uri: ' + id + ' could not be found.')
+  }
+   
+  //grab the 'doc' from the content value space
+  let doc = content.value;
+   
+  // let's just grab the root of the document if its a Document and not a type of Node (ObjectNode or XMLNode)
+  if (doc && (doc instanceof Document || doc instanceof XMLDocument)) {
+    doc = fn.head(doc.root);
+  }
+   
+  //get our instance, default shape of envelope is envelope/instance, else it'll return an empty object/array
+  let instance = datahub.flow.flowUtils.getInstance(doc).toObject() || {};
+   
+  // get triples, return null if empty or cannot be found
+  let triples = datahub.flow.flowUtils.getTriples(doc) || [];
+   
+  //gets headers, return null if cannot be found
+  let headers = datahub.flow.flowUtils.getHeaders(doc) || {};
+   
+  //If you want to set attachments, uncomment here
+  //instance['$attachments'] = doc;
+  instance['$attachments'] = {
+    envelope: {
+      headers: datahub.flow.flowUtils.getHeaders(doc) || {},
+      triples: datahub.flow.flowUtils.getTriples(doc) || [],
+      instance: datahub.flow.flowUtils.getInstance(doc)
+    }
+  };
+   
+   
+  //insert code to manipulate the instance, triples, headers, uri, context metadata, etc.
+     
+  if (instance["Postal"]) {
```

Comment:
This is the only interesting part of the module. I think we need some changes to custom steps so that a developer doesn't need much code besides this.